### PR TITLE
Check `x-ipfs-root-cid` header

### PR DESCRIPTION
**Pull request by peterjan**
# PULL REQUEST

## Overview

This PR checks the response from the `HEAD` call for the `x-ipfs-root-cid` header.
If present we can use it to migrate the CID to Skynet and return a skylink.

## Example for Visual Changes
N/A

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [ ] Testing added or updated for new methods
 - [ ] Verify if any changes impact the WebPortal Health Checks
 - [ ] Approriate documentation updated
 - [ ] Changelog file created

## Issues Closed
N/A

### CODE REVIEW

File: src/services/searchSkylink.js [hunk 1]
```diff
@@ -80,20 +80,6 @@ export default async function searchSkylink(input) {
...
   }
 
-  // any arbitrary url
-  if (input.startsWith("https://")) {
-    const skylink = await requestSkylink(input);
-
-    if (skylink) return skylink;
-  }
-
-  if (input.match(/^[^./]+$/)) {
```
Comment (peterjan) [inline src/services/searchSkylink.js:90]: This was a bit hard to follow, the entire function actually. Since I added the HEAD response check as a fallback I figured it was fine to drop this all together. We explicitly check for hns URLs above using the `HNS_DOMAIN_MATCHER` regex.

File: src/services/searchSkylink.js [hunk 2]
```diff
@@ -80,20 +80,6 @@ export default async function searchSkylink(input) {
     }
   }
 
-  // any arbitrary url
-  if (input.startsWith("https://")) {
```
Comment (peterjan) [inline src/services/searchSkylink.js:84]: I decided to remove this seeing as we perform a HEAD call at the very bottom of the function, so as a last resort, and ignore the error. So in case it wasn't a valid URL there's no harm done really.

File: src/services/searchSkylink.js [hunk 3]
```diff
@@ -107,11 +93,27 @@ export default async function searchSkylink(input) {
 
   if (input.match(IPFS_CID_MATCHER)) {
     const { groups } = input.match(IPFS_CID_MATCHER);
-    const skylink = migrateIpfsToSkylink(groups.cid);
+    const skylink = await migrateIpfsToSkylink(groups.cid);
```
Comment (peterjan) [inline src/services/searchSkylink.js:96]: I guess it's either adding `await`, or just `return migrateIpfsToSkylink(groups.cid);` here. It's redundant to set `skylink` to a promise and then check it because it'll always be evaluated as true and returned.

File: src/services/searchSkylink.js (hunk 1 above)
Comment (peterjan) [reply src/services/searchSkylink.js:90]: I didn't really understand how this regex relates to an hns URL. Which is implied since it's followed up with `const address = await skynetClient.getHnsUrl(input);`

Comment (kwypchlo) [reply src/services/searchSkylink.js:90]: This is to match any string without dots and try to see if it matches hns domain. This allows searches like "uniswap" or "skyfeed".

Comment (peterjan) [reply src/services/searchSkylink.js:90]: ok, brought it back in that case and added a comment

File: src/services/searchSkylink.js [hunk 4]
```diff
@@ -118,9 +120,14 @@ export default async function searchSkylink(input) {
...
   try {
     const response = await ky.head(address, { credentials: "include" });
-    const skylink = response.headers.get("skynet-skylink");
 
+    // check for `skynet-skylink` header
+    const skylink = response.headers.get("skynet-skylink");
     if (skylink) return skylink;
+
+    // check for `x-ipfs-root-cid` header
+    const cid = response.headers.get("skynet-skylink");
```
Comment (peterjan) [inline src/services/searchSkylink.js:129]:
```suggestion
    const cid = response.headers.get("x-ipfs-root-cid");
```

**Comment by kwypchlo** [comment]
due to complications we're closing it for the time being, we might open a different one with other improvements though
